pilot/pkg/model/mesh.go

```diff
@@ -0,0 +1,785 @@
+//     // Create mesh endpoint from relevant values of nativeEp
+//	   meshEndpoints[idx] = NewEndpoint(......)
+//   }
+//   pc.Reconcile(meshEndpoints)
+//
+package model
+
+import (
+	"errors"
+	"fmt"
+	"math"
+	"net"
+	"reflect"
+	"regexp"
+	"sort"
+	"strconv"
+	"sync"
+
+	xdsapi "github.com/envoyproxy/go-control-plane/api"
+	types "github.com/gogo/protobuf/types"
+	multierror "github.com/hashicorp/go-multierror"
+	route "istio.io/api/routing/v1alpha2"
+	"istio.io/istio/pkg/log"
+)
+
+const (
+	// The internal filter name for attributes that should be used for
+	// constructing Istio attributes on the Endpoint in
+	// <a href='https://goo.gl/mFzjpp'>reverse DNS form</a>.
+	istioConfigFilter = "io.istio.istio-config"
+
+	// Key Istio attribute names for mapping endpoints to subsets.
+	// For details, please see
+	// https://istio.io/docs/reference/config/mixer/attribute-vocabulary.html
+
+	// UID is the attribute name for the mesh unique, platform-specific
+	// unique identifier for the server instance of the destination service. Mesh
+	// uses the label to uqniquely identify the endpoint when it receives updates
+	// from the Controller. Example: kubernetes://my-svc-234443-5sffe.my-namespace
+	// No two instances running anywhere in the mesh can have the same value for
+	// UID.
+	UID destinationAttribute = "destination.uid"
+
+	// SERVICE represents the fully qualified __CANONOCAL__ name of the Istio
+	// service, ex: "my-svc.my-namespace.svc.cluster.local".  Mesh uses
+	// this label to identify endpoints belonging to the same service for XDS queries
+	// involving service names. The attribute value is expected to be mesh unique,
+	// meaning services having this name are semantically understood to be the same
+	// modulo version and other labels, irrespective of the platform under which
+	// they are run, the clusters or availability zones they run in.
+	SERVICE destinationAttribute = "destination.service"
+
+	// NAME represents the short part of the service name, ex: "my-svc".
+	NAME destinationAttribute = "destination.name"
+
+	// NAMESPACE represents the namespace part of the destination service,
+	// example: "my-namespace".
+	NAMESPACE destinationAttribute = "destination.namespace"
+
+	// IP represents the IP address of the server instance, example 10.0.0.104.
+	// This IP is expected to be reachable from __THIS__ pilot. No distinction
+	// is being made for directly reachable service instances versus those
+	// behind a VIP. Istio's health discovery service will ensure that this
+	// endpoint's capacity is correctly reported accordingly.
+	IP destinationAttribute = "destination.ip"
+
+	// PORT represents the recipient port on the server IP address, Example: 443
+	PORT destinationAttribute = "destination.port"
+
+	// DOMAIN represents the domain portion of the service name, excluding
+	// the name and namespace, example: svc.cluster.local
+	DOMAIN destinationAttribute = "destination.domain"
```

Comment:
Also remove.. I think all we need is the full name. Every service is supposed to have a hostname. Its the job of the platform adapter to construct the fqdn

Reply:
+1 - a FQDN should be good enough - and we want to support 'bring your name' - so domain doesn't mean what you mean.

Reply:
@costinm and @rshriram again pointing you to  https://istio.io/docs/reference/config/mixer/attribute-vocabulary.html

Reply:
And again pointing you to https://github.com/istio/api/tree/master/routing/v1alpha2 where namespace is absent from inside the spec. Specifically this comment: https://github.com/istio/api/blob/master/routing/v1alpha2/route_rule.proto#L120

Reply:
Another alternative is to drop this as a label if it's not useful to sub-setting. @costinm could you get @ZackButcher and @rshriram to agree?

Reply:
@rshriram  the vocab doc states these to be labels. Is the argument here being made that the namespace label should be prohibited from being used as subset labels? That might actually be a horrrible idea. If a customer has a bunch of K8s clusters along with a mix of other registries like Consul or Eureka, it would be impossible to construct subsets that filter by namespace. Have folks on the steering committee discussed excluding namespaces even for K8s clusters?